node/tools/src/k8s.rs

```diff
@@ -66,145 +219,19 @@ pub async fn create_or_reuse_namespace(client: &Client, name: &str) -> anyhow::R
-                                ..Default::default()
-                            },
-                        ]),
-                        liveness_probe: Some(Probe {
-                            http_get: Some(HTTPGetAction {
-                                path: Some("/health".to_owned()),
-                                port: Int(3154),
-                                ..Default::default()
-                            }),
-                            ..Default::default()
-                        }),
-                        readiness_probe: Some(Probe {
-                            http_get: Some(HTTPGetAction {
-                                path: Some("/health".to_owned()),
-                                port: Int(3154),
-                                ..Default::default()
-                            }),
-                            ..Default::default()
-                        }),
-                        ..Default::default()
-                    }],
-                    ..Default::default()
-                }),
-            },
-            ..Default::default()
-        }),
-        ..Default::default()
-    };
-
-    let deployments: Api<Deployment> = Api::namespaced(client.clone(), namespace);
-    let post_params = PostParams::default();
-    let result = deployments.create(&post_params, &deployment).await?;
-
-    info!(
-        "Deployment: {} , created",
-        result
-            .metadata
-            .name
-            .context("Name not defined in metadata")?
-    );
-    Ok(())
-}
-
-/// Returns a HashMap with mapping: node_id -> IP address
-pub async fn get_seed_node_addrs(
-    client: &Client,
-    amount: usize,
-    namespace: &str,
-) -> anyhow::Result<HashMap<String, String>> {
-    let mut seed_nodes = HashMap::new();
-    let pods: Api<Pod> = Api::namespaced(client.clone(), namespace);
-
-    // Will retry 15 times during 15 seconds to allow pods to start and obtain an IP
-    let retry_strategy = FixedInterval::from_millis(1000).take(15);
-    let pod_list = Retry::spawn(retry_strategy, || get_seed_pods(&pods, amount)).await?;
-
-    for p in pod_list {
-        let node_id = p.labels()["id"].clone();
-        seed_nodes.insert(
-            node_id,
-            p.status
-                .context("Status not present")?
-                .pod_ip
-                .context("Pod IP address not present")?,
-        );
-    }
-    Ok(seed_nodes)
-}
-
-async fn get_seed_pods(pods: &Api<Pod>, amount: usize) -> anyhow::Result<ObjectList<Pod>> {
-    let lp = ListParams::default().labels("seed=true");
-    let p = pods.list(&lp).await?;
-    if p.items.len() == amount && p.iter().all(is_pod_running) {
-        Ok(p)
+async fn get_running_pod(pods: &Api<Pod>, label: &str) -> anyhow::Result<Pod> {
+    let lp = ListParams::default().labels(&format!("app={label}"));
+    let pod = pods
+        .list(&lp)
+        .await?
+        .iter()
+        .next()
+        .context(format!("Pod not found: {label}"))
```

Comment:
Use `with_context(||...)` in case context is not a constant (so that it is not evaluated if not needed)